platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamSingleFileRepairer.java

```diff
@@ -0,0 +1,154 @@
+            throw new IOException("Not able to delete an old repaired file: " + repairedFile.getAbsolutePath());
+        }
+
+        damagedFile = new File(originalFile.getAbsolutePath() + DAMAGED_SUFFIX);
+
+        // Create repairing iterator
+        in = new BufferedInputStream(new FileInputStream(originalFile));
+        repairIterator = new EventStreamSingleFileRepairIterator(new ObjectStreamIterator<>(in, true));
+    }
+
+    /**
+     * Iterates through the input file and writes the possibly repaired event stream to the repaired file name.
+     *
+     * @return true if the written event stream was repaired, false otherwise.
+     * @throws IOException              if a critical exception occurred while traversing the event stream.
+     * @throws NoSuchAlgorithmException if the cryptography is not accessible.
+     */
+    private boolean repairFile() throws IOException, NoSuchAlgorithmException {
+        try (final SerializableDataOutputStream out =
+                new SerializableDataOutputStream(new BufferedOutputStream(new HashingOutputStream(
+                        MessageDigest.getInstance(SHA_384.algorithmName()),
+                        new FileOutputStream(repairedFile, false))))) {
+            for (final int item : EventStreamType.getInstance().getFileHeader()) {
+                out.writeInt(item);
+            }
+            out.writeInt(OBJECT_STREAM_VERSION);
+            while (repairIterator.hasNext()) {
+                out.writeSerializable(repairIterator.next(), true);
+            }
+        }
+        return repairIterator.wasRepaired();
+    }
+
+    /**
+     * Gets the count of the number of DetailedConsensusEvents encountered by the repair iterator.
+     *
+     * @return the number of DetailedConsensusEvents.
+     */
+    public int getEventCount() {
+        return repairIterator.getEventCount();
+    }
+
+    /**
+     * Attempts to repair the event stream file. If the file is repaired, the original file is backed up with a
+     * ".damaged" suffix and replaced with the repaired file. If the original file was not repaired, no change is made.
+     *
+     * @return true if the original event stream file was repaired, false otherwise.
+     * @throws IOException              if a critical exception occurred while traversing the event stream.
+     * @throws NoSuchAlgorithmException if the cryptography is not accessible.
+     */
+    public boolean repair() throws IOException, NoSuchAlgorithmException {
+        final boolean repaired = repairFile();
+
+        try {
+            in.close();
+        } catch (IOException e) {
+            // may have already been closed
+        }
+
+        if (repaired) {
+            if (damagedFile.exists() && !damagedFile.delete()) {
+                throw new IOException("Not able to delete backup damaged file: " + damagedFile.getAbsolutePath());
+            }
+            Files.copy(originalFile.toPath(), damagedFile.toPath());
```

Comment:
Why is it necessary to delete the damaged file and create it again?

Reply:
I will modify it to use the `REPLACE_EXISTING` copy parameter.